### ISSUE TITLE
Broken RFC links in docs

### ISSUE THREAD

**Issue by sanjams2** [awslabs/smithy#1328]
Hi All,

TLDR: Smithy needs to upgrade the Sphinx version used to generate docs to at least `4.4.0` to fix broken rfc links in docs

Details: 

After roaming around the documentation for the project, I noticed that the rfc links present in the docs appear to be broken. As an example the docs for the [`httpHeader ` trait](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpheader-trait) link to an rfc labeled `section 3.2 of RFC7230`. That link is https://tools.ietf.org/html/rfc7230.html#section-3.2. When navigating to this url, a `302 Found` is returned which redirects to https://www.rfc-editor.org/rfc/rfc7230.html.html which returns a `404 Not Found`. 

I realize that this link is generated by Sphinx and after digging around in their repo I found an [issue](https://github.com/sphinx-doc/sphinx/issues/9857) raised for this and a [subsequent fix](https://github.com/sphinx-doc/sphinx/pull/9858) to change the url Sphinx generates. This fix was [released in Sphinx 4.4.0](https://www.sphinx-doc.org/en/master/changes.html#release-4-4-0-released-jan-17-2022). However, Smithy [locks its Sphinx version](https://github.com/awslabs/smithy/blob/a7a16dfd13426f712934108f46cce3a3bb9d61b4/docs/requirements.txt#L1) between `4.2.0` and `4.3.0`. So it seems Smithy will need to upgrade the Sphinx version to at least `4.4.0` to fix the broken links.

**Comment by srchase** [comment]
Following the release of IDL 2.0, the docs received a major update.

In both the 1.0 and 2.0 spec pages, the `section 3.2 of RFC7230` link seems to be working as expected:
[httpHeader on 2.0](https://awslabs.github.io/smithy/2.0/spec/http-bindings.html#httpheader-trait)
[httpHeader on 1.0](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpheader-trait)

Since those are working, the change in #1329 is not needed to fix this issue.

**Comment by sanjams2** [comment]
Strange. For my own curiosity, when was that release pushed? It also now appears that the original redirect link is also fixed too.
